pytradfri/resource.py

```diff
@@ -78,8 +78,7 @@ def observe_callback(value: TypeRaw) -> None:
             """
             self.raw = self._model_class(**value)
 
-            if callback:
-                callback(self)
+            callback(self)
```

Comment:
Mypy returned `pytradfri/resource.py:81: error: Function "Callable[[ApiResource], None]" could always be true in boolean context  [truthy-function]`

Looks like the if-clause was introduced as a part of the early stage typing efforts here and shouldn't be there?

Reply:
Have we accepted None as callback ever?

Reply:
Oh, I think we used to:
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/21142447/201761860-b33f6aab-c492-4f9b-bf5f-1da168d94a08.png">

Reply:
https://github.com/home-assistant-libs/pytradfri/blob/2fc8fa0ec1a639a0ace6093b1311d2009d453123/pytradfri/command.py

Reply:
Then we should probably keep the check here and extend the type annotation to be optional.